### ISSUE TITLE
♻️ Convert OAI feed to JSON document to have one row representing a file set

### ISSUE THREAD

**Issue by jeremyf** [scientist-softserv/adventist_knapsack#398]
Depends on:

-   scientist-softserv/adventist_knapsack#420

At present the generated CSV from scientist-softserv/adventist_knapsack#420 has one row per file.  What we need is one row per file set.  That is to say, each file that we encounter in the OAI feed is not it's own FileSet.  Instead there is a relationship between those files.

For example we might have an Archival PDF, a Reader PDF, extract text, and a thumbnail.

What we need is the following somewhat arbitrary CSV.  Each row will have the following columns:

-   OAI set (the OAI set in which we found this item)
-   Work Identifier (e.g. aark\_id)
-   Original

Additional columns could be:

-   text : points to the file that has the extracted
-   thumbnail : points to the file that has the thumbnail
-   reader : points to the reader

Note, there are likely files that do not fall into any of these categories.  We won't know what these are without data analysis of the output.

Initial proposal:

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">

<colgroup>
<col  class="org-left" />

<col  class="org-right" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">OAI_SET</th>
<th scope="col" class="org-right">AARK_ID</th>
<th scope="col" class="org-left">ORIGINAL</th>
<th scope="col" class="org-left">TEXT</th>
<th scope="col" class="org-left">READER</th>
<th scope="col" class="org-left">THUMBNAIL</th>
<th scope="col" class="org-left">&#xa0;</th>
<th scope="col" class="org-left">&#xa0;</th>
<th scope="col" class="org-left">&#xa0;</th>
</tr>
</thead>

<tbody>
<tr>
<td class="org-left">adl:periodicals</td>
<td class="org-right">1234</td>
<td class="org-left">1234.archival.pdf</td>
<td class="org-left">1234.txt</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">1234.thumbnail.png</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">adl:article</td>
<td class="org-right">1235</td>
<td class="org-left">1234.obj.tiff</td>
<td class="org-left">1234.txt</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">1234.thumbnail.png</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">adl:periodicals</td>
<td class="org-right">1236</td>
<td class="org-left">1236.archival.pdf</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">1236.reader.pdf</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">1236.other.file.fmt</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>

In other words, we’ll have a sparsely populated CSV; some cells will be empty, and we’ll possibly have untitled columns representing unknown files.

The first step is to understand the types of files we might have.  And do our best to categorize them into their appropriate location.

**Note:** This might be easier to write as JSON, but JSON is terrible for human reading.
